codegen/cmd/injection-gen/generators/reconciler_reconciler.go

```diff
@@ -429,21 +435,21 @@ func (r *reconcilerImpl) Reconcile(ctx {{.contextContext|raw}}, key string) erro
 		if resource, err = r.setFinalizerIfFinalizer(ctx, resource); err != nil {
 			return {{.fmtErrorf|raw}}("failed to set finalizers: %w", err)
 		}
-		{{if .isKRShaped}}
+		{{if .hasStatus}}{{if .isKRShaped}}
```

Comment:
Doesn't KRShaped include status? Feels like a redundant check.

Reply:
What `isKRShaped` means in first instance? :smiley:

Reply:
https://github.com/knative/pkg/blob/0d31134d5f4e37a2d44392ec8fb8790e2dd4f7e7/apis/duck/v1/kresource_type.go#L31-L39

Reply:
Thanks, changed!